### ISSUE TITLE
Remove HMR feature

### ISSUE THREAD

**Issue by DimitryDushkin** [react/metro#336]
1) HMR is quite broken for new React releases. Check out number of issues in it https://github.com/gaearon/react-hot-loader/issues  and this https://github.com/gaearon/react-hot-loader/issues/1024
2) This library is not evolving. It's not supporting forwardRef, hooks, lazy.
3) It's wrapping every component in react-proxy, which complicates debugging and also not compatable with class-based components without transpilation to functions.
4) Takes lots of time to make it work with redux-based applications

So I suggest to remove it from metro.

**Comment by theKashey** [comment]
The problem is - "metro" is using React-Hot-Loader v2. Right now it's version 4, and version 5 is planned to fix [1024](
https://github.com/gaearon/react-hot-loader/issues/1024), and probably any other issue.

It's better to upgrade it, than to remove.

**Comment by theKashey** [comment]
True. Current HMR solution is a bit _aggresive_. Even _offensive_.

**Comment by robhogan** [comment]
For posterity - HMR was replaced by Fast Refresh in https://github.com/react/metro/releases/v0.55.0 / RN 0.61
